web/src/users/add_url.rs

```diff
@@ -18,16 +26,39 @@ pub async fn insert_article_handler(
     State(state): State<AppState>,
     user: User,
     Form(form): Form<ArticleForm>,
-) -> impl IntoResponse {
+) -> WebResult<impl IntoResponse> {
+    info!("Received request to insert article: {}", form.url); // Log the received URL
+
     let url = form.url;
     let user_id = user.user_id;
 
-    tracing::info!("{}", url);
-    match add_url(&state.db, &url, user_id, &true).await {
-        Ok(_) => Redirect::to("/dashboard?flash[success]=Article added"),
-        Err(e) => {
-            eprintln!("Error adding URL: {e:?}");
-            Redirect::to("/dashboard?flash[error]=Could not add article")
-        }
-    }
+    let page = add_url(&state.db, &url, user_id, &true).await?;
+    let page_id = page.page().page_id;
+
+    let raw_html = download_raw_html(&url).await?;
+    let url_parsed = Url::parse(&url)?;
+    let cleaned_html = clean_raw_html(&raw_html, &url_parsed)?;
+
+    let (markdown, _) = store_markdown(&state.db, page_id, &cleaned_html)
+        .await
+        .unwrap();
+    let markdown_id = markdown.markdown_id;
+
+    generate_and_store_summary(&state.db, markdown_id, &cleaned_html)
+        .await
+        .unwrap();
```

Comment:
All of this is what happens in the `ProcessArticle` job right?

I don't think we want to do it in the Handler, we should just do all this in the job!

The reason for that is we don't want to do all this work in the web request to add the URL, we want to do it in the 'background' which is what our job helps us do!